ReactViewControl/ReactViewRender.cs

```diff
@@ -187,15 +187,14 @@ private enum LoadStatus {
             ExecuteLoaderFunction("loadStyleSheet", frameName, loadArg);
         }
 
-        private string GetMappings() {
-            return JavascriptSerializer.Serialize(Plugins.Select(m => new KeyValuePair<string, object>(m.Name, NormalizeUrl(ToFullUrl(m.JavascriptSource)))));
-        }
-
         private void InternalLoadPlugins(string frameName) {
             var pluginsWithNativeObject = Plugins.Where(p => !string.IsNullOrEmpty(p.NativeObjectName)).ToArray();
             var loadArgs = new[] {
-                JavascriptSerializer.Serialize(pluginsWithNativeObject.Select(m => new[] { m.Name, GetNativeObjectFullName(m.NativeObjectName, frameName) })), // plugins
-                GetMappings()
+                JavascriptSerializer.Serialize(pluginsWithNativeObject.Select(m => new [] {
+                    m.Name,
+                    GetNativeObjectFullName(m.NativeObjectName, frameName),
+                    ToFullUrl(NormalizeUrl(m.MainSource)),
+                }.Concat(m.ExternalSources.Select(s => ToFullUrl(NormalizeUrl(s))))))
```

Comment:
By the time you load plugins you still don't have the ExternalSources ready... this has to be loaded when the component is loaded not with the plugins

Reply:
These are the "dependency sources" of the plugin, i.e. shared chunks might be generated even for plugins

Reply:
oh ... you're right, I misread the code